src/inmanta/deploy/persistence.py

```diff
@@ -289,3 +319,65 @@ def get_resource_action_logger(self, environment: UUID) -> logging.Logger:
         :return: The logger for the given environment.
         """
         return self._resource_action_logger.getChild(str(environment))
+
+    async def dryrun_update(self, env: UUID, dryrun_result: DryrunResult) -> None:
+        # async with self.dryrun_lock: Should this class have this lock or one of the child classes?
```

Comment:
How do we proceed with this lock? Is it needed

Reply:
It turns out we do need it, because the dryrun table is not very well structured.